htdocs/mrp/class/mo.class.php

```diff
@@ -694,6 +694,20 @@ public function updateProduction(User $user, $notrigger = true)
 							$moline->fk_mo = $this->id;
 							$moline->origin_id = $line->id;
 							$moline->origin_type = 'bomline';
+
+							//-- convert to second related to c_unit table
+							$tmpproduct = new Product($this->db);
+							$res = $tmpproduct->fetch($line->fk_product);
+							if ($res && $tmpproduct->isService() && isModEnabled('stock') && !getDolGlobalInt('STOCK_SUPPORTS_SERVICES')) {
+								$sql = "SELECT scale FROM ".$this->db->prefix()."c_units WHERE rowid =". (int) $line->fk_unit;
+								$resql = $this->db->query($sql);
+								if ($resql) {
+									$objTmp = $this->db->fetch_object($resql);
+									$line->qty = $line->qty * $objTmp->scale;
```

Comment:
I don't understand the goal of this.
If qty in line is saved as 10, it must be reported as 10 on screen.